src/views/OrderDetailUpdated.vue

```diff
@@ -569,52 +569,43 @@ export default defineComponent({
       }
 
       let order = JSON.parse(JSON.stringify(this.order))
-
-      // https://blog.devgenius.io/using-async-await-in-a-foreach-loop-you-cant-c174b31999bd
-      // The forEach, map, reduce loops are not built to work with asynchronous callback functions.
-      // It doesn't wait for the promise of an iteration to be resolved before it goes on to the next iteration.
-      // We could use either the for…of the loop or the for(let i = 0;….)
+      const isEntireOrderRejection = this.isEntierOrderRejectionEnabled(order);
+      const rejectToFacilityId = order.part.shipmentMethodEnum.shipmentMethodEnumId === "STOREPICKUP" ? "PICKUP_REJECTED" : null;
+      const itemsToReject: any[] = [];
+      
       for (const item of order.part.items) {
-        let params = {} as any;
-        if(this.isEntierOrderRejectionEnabled(order)) {
-          params = {
-            ...payload,
-            rejectReason: item.rejectReason || this.rejectEntireOrderReasonId,
-            facilityId: item.facilityId,
+        const shouldReject = isEntireOrderRejection || item.rejectReason;
+
+        if (shouldReject) {
+          itemsToReject.push({
             orderItemSeqId: item.orderItemSeqId,
-            shipmentMethodTypeId: order.part.shipmentMethodEnum.shipmentMethodEnumId,
             quantity: parseInt(item.quantity),
-            ...(order.part.shipmentMethodEnum.shipmentMethodEnumId === "STOREPICKUP" && ({"naFacilityId": "PICKUP_REJECTED"})),
-          }
-        } else if(item.rejectReason) {
-          params = {
-            ...payload,
-            rejectReason: item.rejectReason || this.rejectEntireOrderReasonId,
             facilityId: item.facilityId,
-            orderItemSeqId: item.orderItemSeqId,
-            shipmentMethodTypeId: order.part.shipmentMethodEnum.shipmentMethodEnumId,
-            quantity: parseInt(item.quantity),
-            ...(order.part.shipmentMethodEnum.shipmentMethodEnumId === "STOREPICKUP" && ({"naFacilityId": "PICKUP_REJECTED"})),
-          }
+            updateQOH: false, // Could be true if QOH needs to be updated on rejection
+            rejectionReasonId: item.rejectReason || this.rejectEntireOrderReasonId,
+            kitComponents: isKit(item) ? item.rejectedComponents || [] : []
```

Comment:
We need to pass shipmentMethodTypeId with item so that the rejectOrderItem service handles the facility properly.